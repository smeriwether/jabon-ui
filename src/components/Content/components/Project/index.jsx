import React, { Component } from 'react';
import PropTypes from 'prop-types';

import ExpandDown from '../../../../UI/ExpandDown';

const propTypes = {
  project: PropTypes.shape({
    wsdlLink: PropTypes.string,
    header1: PropTypes.shape({
      key: PropTypes.string,
      value: PropTypes.string
    }),
    header2: PropTypes.shape({
      key: PropTypes.string,
      value: PropTypes.string
    })
  }),
  onChange: PropTypes.func.isRequired,
  onExpand: PropTypes.func.isRequired
};

const defaultProps = {
  project: {
    wsdlLink: '',
    header1: { key: '', value: '' },
    header2: { key: '', value: '' }
  }
};

const styles = {
  label: 'fw7 f6',
  input: 'w-100 pa2 mt2 br2 b--black-20 ba f6'
};

const text = {
  placeholder: 'Placeholder...'
};

class Project extends Component {
  constructor(props) {
    super(props);

    this.state = {
      wsdlLink: props.project.wsdlLink,
      header1Key: props.project.header1.key,
      header1Value: props.project.header1.value,
      header2Key: props.project.header2.key,
      header2Value: props.project.header2.value
    };
  }

  onWsdlChange = (input) => {
    const value = input.target.value;
    this.setState({ wsdlLink: value });
    this.props.onChange({ wsdlLink: value });
  }

  onHeader1KeyChange = (input) => {
    const value = input.target.value;
    this.setState({ header1Key: value });
    this.props.onChange({ header1: { key: value, value: this.state.header1Value } });
  }

  onHeader1ValueChange = (input) => {
    const value = input.target.value;
    this.setState({ header1Value: value });
    this.props.onChange({ header1: { key: this.state.header1Key, value: value } });
  }

  onHeader2KeyChange = (input) => {
    const value = input.target.value;
    this.setState({ header2Key: value });
    this.props.onChange({ header2: { key: value, value: this.state.header2Value } });
  }

  onHeader2ValueChange = (input) => {
    const value = input.target.value;
    this.setState({ header2Value: value });
    this.props.onChange({ header2: { key: this.state.header2Key, value: value } });
  }

  render() {
    return (
      <div className='h-100 w-100 pa3 near-black overflow-scroll'>
        <div className='flex justify-between'>
          <div className='b f3'>
            Project Information
          </div>
          <div className='pointer' data-role='expand' onClick={ this.props.onExpand } >
            <ExpandDown />
          </div>
        </div>

        <div className='pt4 pb4'>
          <label className={ styles.label } htmlFor='wsdl'>WSDL Link</label>
          <input
            id='wsdl'
            data-role='wsdl'
            value={ this.state.wsdlLink }
            className={ styles.input }
            placeholder={ text.placeholder }
            onChange={ (i) => { this.onWsdlChange(i) } }
          >
          </input>
        </div>

        <div className='w-50 fl pr3 pb3'>
          <label className={ styles.label } htmlFor='header1-key'>Header Key</label>
          <input
            id='header1-key'
            data-role='header1-key'
            value={ this.state.header1Key }
            className={ styles.input }
            placeholder={ text.placeholder }
            onChange={ (i) => { this.onHeader1KeyChange(i) } }
          >
          </input>
        </div>
        <div className='w-50 fl pb3'>
          <label className={ styles.label } htmlFor='header1-value'>Header Value</label>
          <input
            id='header1-value'
            data-role='header1-value'
            value={ this.state.header1Value }
            className={ styles.input }
            placeholder={ text.placeholder }
            onChange={ (i) => { this.onHeader1ValueChange(i) } }
          >
          </input>
        </div>

        <div className='w-50 fl pr3 pb3'>
          <label className={ styles.label } htmlFor='header2-key'>Header Key</label>
          <input
            id='header2-key'
            data-role='header2-key'
            value={ this.state.header2Key }
            className={ styles.input }
            placeholder={ text.placeholder }
            onChange={ (i) => { this.onHeader2KeyChange(i) } }
          >
          </input>
        </div>
        <div className='w-50 fl pb3'>
          <label className={ styles.label } htmlFor='header2-value'>Header Value</label>
          <input
            id='header2-value'
            data-role='header2-value'
            value={ this.state.header2Value }
            className={ styles.input }
            placeholder={ text.placeholder }
            onChange={ (i) => { this.onHeader2ValueChange(i) } }
          >
          </input>
        </div>
      </div>
    );
  }
};

Project.propTypes = propTypes;
Project.defaultProps = defaultProps;

export default Project;
