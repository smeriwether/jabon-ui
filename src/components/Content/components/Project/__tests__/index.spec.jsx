import Project from '../../Project';

describe('<Project />', () => {
  const setup = (overrides) => {
    const props = {
      project: {
        wsdlLink: '',
        header1: { key: '', value: '' },
        header2: { key: '', value: '' }
      },
      onChange: jest.fn(),
      onExpand: jest.fn(),
      ...overrides
    };

    const wrapper = shallow(<Project { ...props } />);

    return { wrapper };
  };

  it('should match the snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('sets the default values from props', () => {
    const { wrapper } = setup({
      project: {
        wsdlLink: 'http://example.com',
        header1: { key: 'key1', value: 'value1' },
        header2: { key: 'key2', value: 'value2' },
      }
    });

    expect(wrapper.find('input[data-role="wsdl"]').prop('value')).toEqual('http://example.com');
    expect(wrapper.find('input[data-role="header1-key"]').prop('value')).toEqual('key1');
    expect(wrapper.find('input[data-role="header1-value"]').prop('value')).toEqual('value1');
    expect(wrapper.find('input[data-role="header2-key"]').prop('value')).toEqual('key2');
    expect(wrapper.find('input[data-role="header2-value"]').prop('value')).toEqual('value2');
  });

  context('when the wsdl input is changed', () => {
    it('should call the onChange prop with the wsdl update', () => {
      const onChangeMock = jest.fn();
      const { wrapper } = setup({ onChange: onChangeMock });

      const wsdlLink = 'http://example.com';
      wrapper.find('input[data-role="wsdl"]').simulate('change', { target: { value: wsdlLink } });

      expect(onChangeMock).toHaveBeenCalled();
      expect(onChangeMock).toHaveBeenCalledWith({ wsdlLink });
    });
  });

  context('when the header 1 inputs are changed', () => {
    it('should call the onChange prop with the update', () => {
      const onChangeMock = jest.fn();
      const { wrapper } = setup({ onChange: onChangeMock });

      const headerKey = 'Authorization';
      wrapper.find('input[data-role="header1-key"]').simulate('change', { target: { value: headerKey } });

      expect(onChangeMock).toHaveBeenCalled();
      expect(onChangeMock).toHaveBeenCalledWith({ header1: { key: headerKey, value: '' } });

      const headerValue = 'some-auth-key';
      wrapper.find('input[data-role="header1-value"]').simulate('change', { target: { value: headerValue } });

      expect(onChangeMock).toHaveBeenCalled();
      expect(onChangeMock).toHaveBeenCalledWith({ header1: { key: headerKey, value: headerValue } });
    });
  });

  context('when the header 2 inputs are changed', () => {
    it('should call the onChange prop with the update', () => {
      const onChangeMock = jest.fn();
      const { wrapper } = setup({ onChange: onChangeMock });

      const headerValue = 'some-auth-key';
      wrapper.find('input[data-role="header2-value"]').simulate('change', { target: { value: headerValue } });

      expect(onChangeMock).toHaveBeenCalled();
      expect(onChangeMock).toHaveBeenCalledWith({ header2: { key: '', value: headerValue } });

      const headerKey = 'Authorization';
      wrapper.find('input[data-role="header2-key"]').simulate('change', { target: { value: headerKey } });

      expect(onChangeMock).toHaveBeenCalled();
      expect(onChangeMock).toHaveBeenCalledWith({ header2: { key: headerKey, value: headerValue } });
    });
  });

  context('when expand button is clicked', () => {
    it('should call the onExpand prop', () => {
      const onExpandMock = jest.fn();
      const { wrapper } = setup({ onExpand: onExpandMock });

      wrapper.find('div[data-role="expand"]').simulate('click');

      expect(onExpandMock).toHaveBeenCalled();
    });
  });
});
