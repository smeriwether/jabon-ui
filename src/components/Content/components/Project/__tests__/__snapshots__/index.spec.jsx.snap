// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Project /> should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Project
    onChange={[MockFunction]}
    onExpand={[MockFunction]}
    project={
      Object {
        "header1": Object {
          "key": "",
          "value": "",
        },
        "header2": Object {
          "key": "",
          "value": "",
        },
        "wsdlLink": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="flex justify-between"
        >
          <div
            className="b f3"
          >
            Project Information
          </div>
          <div
            className="pointer"
            data-role="expand"
            onClick={[MockFunction]}
          >
            <ExpandDown
              fill="#000000"
            />
          </div>
        </div>,
        <div
          className="pt4 pb4"
        >
          <label
            className="fw7 f6"
            htmlFor="wsdl"
          >
            WSDL Link
          </label>
          <input
            className="w-100 pa2 mt2 br2 b--black-20 ba f6"
            data-role="wsdl"
            id="wsdl"
            onChange={[Function]}
            placeholder="Placeholder..."
            value=""
          />
        </div>,
        <div
          className="w-50 fl pr3 pb3"
        >
          <label
            className="fw7 f6"
            htmlFor="header1-key"
          >
            Header Key
          </label>
          <input
            className="w-100 pa2 mt2 br2 b--black-20 ba f6"
            data-role="header1-key"
            id="header1-key"
            onChange={[Function]}
            placeholder="Placeholder..."
            value=""
          />
        </div>,
        <div
          className="w-50 fl pb3"
        >
          <label
            className="fw7 f6"
            htmlFor="header1-value"
          >
            Header Value
          </label>
          <input
            className="w-100 pa2 mt2 br2 b--black-20 ba f6"
            data-role="header1-value"
            id="header1-value"
            onChange={[Function]}
            placeholder="Placeholder..."
            value=""
          />
        </div>,
        <div
          className="w-50 fl pr3 pb3"
        >
          <label
            className="fw7 f6"
            htmlFor="header2-key"
          >
            Header Key
          </label>
          <input
            className="w-100 pa2 mt2 br2 b--black-20 ba f6"
            data-role="header2-key"
            id="header2-key"
            onChange={[Function]}
            placeholder="Placeholder..."
            value=""
          />
        </div>,
        <div
          className="w-50 fl pb3"
        >
          <label
            className="fw7 f6"
            htmlFor="header2-value"
          >
            Header Value
          </label>
          <input
            className="w-100 pa2 mt2 br2 b--black-20 ba f6"
            data-role="header2-value"
            id="header2-value"
            onChange={[Function]}
            placeholder="Placeholder..."
            value=""
          />
        </div>,
      ],
      "className": "h-100 w-100 pa3 near-black overflow-scroll",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="b f3"
            >
              Project Information
            </div>,
            <div
              className="pointer"
              data-role="expand"
              onClick={[MockFunction]}
            >
              <ExpandDown
                fill="#000000"
              />
            </div>,
          ],
          "className": "flex justify-between",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Project Information",
              "className": "b f3",
            },
            "ref": null,
            "rendered": "Project Information",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ExpandDown
                fill="#000000"
              />,
              "className": "pointer",
              "data-role": "expand",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "fill": "#000000",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="fw7 f6"
              htmlFor="wsdl"
            >
              WSDL Link
            </label>,
            <input
              className="w-100 pa2 mt2 br2 b--black-20 ba f6"
              data-role="wsdl"
              id="wsdl"
              onChange={[Function]}
              placeholder="Placeholder..."
              value=""
            />,
          ],
          "className": "pt4 pb4",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "WSDL Link",
              "className": "fw7 f6",
              "htmlFor": "wsdl",
            },
            "ref": null,
            "rendered": "WSDL Link",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "w-100 pa2 mt2 br2 b--black-20 ba f6",
              "data-role": "wsdl",
              "id": "wsdl",
              "onChange": [Function],
              "placeholder": "Placeholder...",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="fw7 f6"
              htmlFor="header1-key"
            >
              Header Key
            </label>,
            <input
              className="w-100 pa2 mt2 br2 b--black-20 ba f6"
              data-role="header1-key"
              id="header1-key"
              onChange={[Function]}
              placeholder="Placeholder..."
              value=""
            />,
          ],
          "className": "w-50 fl pr3 pb3",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Header Key",
              "className": "fw7 f6",
              "htmlFor": "header1-key",
            },
            "ref": null,
            "rendered": "Header Key",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "w-100 pa2 mt2 br2 b--black-20 ba f6",
              "data-role": "header1-key",
              "id": "header1-key",
              "onChange": [Function],
              "placeholder": "Placeholder...",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="fw7 f6"
              htmlFor="header1-value"
            >
              Header Value
            </label>,
            <input
              className="w-100 pa2 mt2 br2 b--black-20 ba f6"
              data-role="header1-value"
              id="header1-value"
              onChange={[Function]}
              placeholder="Placeholder..."
              value=""
            />,
          ],
          "className": "w-50 fl pb3",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Header Value",
              "className": "fw7 f6",
              "htmlFor": "header1-value",
            },
            "ref": null,
            "rendered": "Header Value",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "w-100 pa2 mt2 br2 b--black-20 ba f6",
              "data-role": "header1-value",
              "id": "header1-value",
              "onChange": [Function],
              "placeholder": "Placeholder...",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="fw7 f6"
              htmlFor="header2-key"
            >
              Header Key
            </label>,
            <input
              className="w-100 pa2 mt2 br2 b--black-20 ba f6"
              data-role="header2-key"
              id="header2-key"
              onChange={[Function]}
              placeholder="Placeholder..."
              value=""
            />,
          ],
          "className": "w-50 fl pr3 pb3",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Header Key",
              "className": "fw7 f6",
              "htmlFor": "header2-key",
            },
            "ref": null,
            "rendered": "Header Key",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "w-100 pa2 mt2 br2 b--black-20 ba f6",
              "data-role": "header2-key",
              "id": "header2-key",
              "onChange": [Function],
              "placeholder": "Placeholder...",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="fw7 f6"
              htmlFor="header2-value"
            >
              Header Value
            </label>,
            <input
              className="w-100 pa2 mt2 br2 b--black-20 ba f6"
              data-role="header2-value"
              id="header2-value"
              onChange={[Function]}
              placeholder="Placeholder..."
              value=""
            />,
          ],
          "className": "w-50 fl pb3",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Header Value",
              "className": "fw7 f6",
              "htmlFor": "header2-value",
            },
            "ref": null,
            "rendered": "Header Value",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "w-100 pa2 mt2 br2 b--black-20 ba f6",
              "data-role": "header2-value",
              "id": "header2-value",
              "onChange": [Function],
              "placeholder": "Placeholder...",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="flex justify-between"
          >
            <div
              className="b f3"
            >
              Project Information
            </div>
            <div
              className="pointer"
              data-role="expand"
              onClick={[MockFunction]}
            >
              <ExpandDown
                fill="#000000"
              />
            </div>
          </div>,
          <div
            className="pt4 pb4"
          >
            <label
              className="fw7 f6"
              htmlFor="wsdl"
            >
              WSDL Link
            </label>
            <input
              className="w-100 pa2 mt2 br2 b--black-20 ba f6"
              data-role="wsdl"
              id="wsdl"
              onChange={[Function]}
              placeholder="Placeholder..."
              value=""
            />
          </div>,
          <div
            className="w-50 fl pr3 pb3"
          >
            <label
              className="fw7 f6"
              htmlFor="header1-key"
            >
              Header Key
            </label>
            <input
              className="w-100 pa2 mt2 br2 b--black-20 ba f6"
              data-role="header1-key"
              id="header1-key"
              onChange={[Function]}
              placeholder="Placeholder..."
              value=""
            />
          </div>,
          <div
            className="w-50 fl pb3"
          >
            <label
              className="fw7 f6"
              htmlFor="header1-value"
            >
              Header Value
            </label>
            <input
              className="w-100 pa2 mt2 br2 b--black-20 ba f6"
              data-role="header1-value"
              id="header1-value"
              onChange={[Function]}
              placeholder="Placeholder..."
              value=""
            />
          </div>,
          <div
            className="w-50 fl pr3 pb3"
          >
            <label
              className="fw7 f6"
              htmlFor="header2-key"
            >
              Header Key
            </label>
            <input
              className="w-100 pa2 mt2 br2 b--black-20 ba f6"
              data-role="header2-key"
              id="header2-key"
              onChange={[Function]}
              placeholder="Placeholder..."
              value=""
            />
          </div>,
          <div
            className="w-50 fl pb3"
          >
            <label
              className="fw7 f6"
              htmlFor="header2-value"
            >
              Header Value
            </label>
            <input
              className="w-100 pa2 mt2 br2 b--black-20 ba f6"
              data-role="header2-value"
              id="header2-value"
              onChange={[Function]}
              placeholder="Placeholder..."
              value=""
            />
          </div>,
        ],
        "className": "h-100 w-100 pa3 near-black overflow-scroll",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="b f3"
              >
                Project Information
              </div>,
              <div
                className="pointer"
                data-role="expand"
                onClick={[MockFunction]}
              >
                <ExpandDown
                  fill="#000000"
                />
              </div>,
            ],
            "className": "flex justify-between",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Project Information",
                "className": "b f3",
              },
              "ref": null,
              "rendered": "Project Information",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ExpandDown
                  fill="#000000"
                />,
                "className": "pointer",
                "data-role": "expand",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "fill": "#000000",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="fw7 f6"
                htmlFor="wsdl"
              >
                WSDL Link
              </label>,
              <input
                className="w-100 pa2 mt2 br2 b--black-20 ba f6"
                data-role="wsdl"
                id="wsdl"
                onChange={[Function]}
                placeholder="Placeholder..."
                value=""
              />,
            ],
            "className": "pt4 pb4",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "WSDL Link",
                "className": "fw7 f6",
                "htmlFor": "wsdl",
              },
              "ref": null,
              "rendered": "WSDL Link",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "w-100 pa2 mt2 br2 b--black-20 ba f6",
                "data-role": "wsdl",
                "id": "wsdl",
                "onChange": [Function],
                "placeholder": "Placeholder...",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="fw7 f6"
                htmlFor="header1-key"
              >
                Header Key
              </label>,
              <input
                className="w-100 pa2 mt2 br2 b--black-20 ba f6"
                data-role="header1-key"
                id="header1-key"
                onChange={[Function]}
                placeholder="Placeholder..."
                value=""
              />,
            ],
            "className": "w-50 fl pr3 pb3",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Header Key",
                "className": "fw7 f6",
                "htmlFor": "header1-key",
              },
              "ref": null,
              "rendered": "Header Key",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "w-100 pa2 mt2 br2 b--black-20 ba f6",
                "data-role": "header1-key",
                "id": "header1-key",
                "onChange": [Function],
                "placeholder": "Placeholder...",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="fw7 f6"
                htmlFor="header1-value"
              >
                Header Value
              </label>,
              <input
                className="w-100 pa2 mt2 br2 b--black-20 ba f6"
                data-role="header1-value"
                id="header1-value"
                onChange={[Function]}
                placeholder="Placeholder..."
                value=""
              />,
            ],
            "className": "w-50 fl pb3",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Header Value",
                "className": "fw7 f6",
                "htmlFor": "header1-value",
              },
              "ref": null,
              "rendered": "Header Value",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "w-100 pa2 mt2 br2 b--black-20 ba f6",
                "data-role": "header1-value",
                "id": "header1-value",
                "onChange": [Function],
                "placeholder": "Placeholder...",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="fw7 f6"
                htmlFor="header2-key"
              >
                Header Key
              </label>,
              <input
                className="w-100 pa2 mt2 br2 b--black-20 ba f6"
                data-role="header2-key"
                id="header2-key"
                onChange={[Function]}
                placeholder="Placeholder..."
                value=""
              />,
            ],
            "className": "w-50 fl pr3 pb3",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Header Key",
                "className": "fw7 f6",
                "htmlFor": "header2-key",
              },
              "ref": null,
              "rendered": "Header Key",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "w-100 pa2 mt2 br2 b--black-20 ba f6",
                "data-role": "header2-key",
                "id": "header2-key",
                "onChange": [Function],
                "placeholder": "Placeholder...",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="fw7 f6"
                htmlFor="header2-value"
              >
                Header Value
              </label>,
              <input
                className="w-100 pa2 mt2 br2 b--black-20 ba f6"
                data-role="header2-value"
                id="header2-value"
                onChange={[Function]}
                placeholder="Placeholder..."
                value=""
              />,
            ],
            "className": "w-50 fl pb3",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Header Value",
                "className": "fw7 f6",
                "htmlFor": "header2-value",
              },
              "ref": null,
              "rendered": "Header Value",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "w-100 pa2 mt2 br2 b--black-20 ba f6",
                "data-role": "header2-value",
                "id": "header2-value",
                "onChange": [Function],
                "placeholder": "Placeholder...",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
